docs/README_API_development_workflow.md

```diff
@@ -0,0 +1,109 @@
+# API - development workflow
+
+This document outlines a proposed general workflow for the development of
+features in the API and GeoProcessing services.
+
+As the development team scales out, it is desirable to follow (flexibly) a
+workflow that allows to optimize clarity and parallel work.
+
+## Checklist
+
+### Alignment on requirements for a component
+
+* are all the project team members aware of the user-value of the component?
+* are there any assumptions implicit in the design that need clarification or
+  adjustment (because of product needs or because of technical or other
+  limitations)?
+
+### definition of the API interface(s)
+
+* which "resources" (in a REST meaning) need to be handled?
+* which endpoints are needed?
+  * should any *existing* endpoint be changed?
+  * are *new* endpoints needed?
+  * should the new endpoints share part of the URL path with existing
+    endpoints? (this is partly related to a section below on fronting
+    implementation details with a simpler public interface)
+* at this stage we should create all the relevant DTOs for create, update and
+  get payloads (requests and responses), as well as all relevant types
+* if relevant, Swagger documentation may be added now (or after the stub
+  implementation, see below)
+  
+At this stage we should have full typing as well as scaffolded endpoints. These
+would do little by now, except validating payload shape.
+
+### API e2e tests
+
+* we should aim to add at least happy-path e2e tests as one of the first
+  implementation steps
+* this means that we should create e2e tests that mimic real requests the API
+  may receive from API clients, starting from the frontend app
+* e2e tests should use DTOs and types from the previous step
+
+At this stage, we should be able to run tests for the new component and see them
+fail.
+
+### Stub implementation
+
+* consider whether a CRUD implementation is sufficient for the task
+* if so, and if queries require JSON:API query features (filtering, sorting,
+  sparse fieldsets, includes, pagination), we should base a new service on
+  `AppBaseService`
+* if the CQRS model may be more appropriate, there's still an option to go
+  very lightweight on this by only keeping command and query DTOs fully distinct
+* if CQRS proper is appropriate, it may be the way to go
```

Comment:
It may be just a common facade instead of CQRS if we don't plan cross-module interactions (i.e. everything happens within the same module).

Example of existing (PR) CQRS usage comes from the fact that update on scenario triggers some processing in different area.